### ISSUE TITLE
dataPath is empty for dependencies

### ISSUE THREAD

**Issue by jsdmc** [ajv-validator/ajv#68]
Hi there) 
I have some simple schema, let's say

``` javascript
var schema = {
  "type": "object",
  "properties": {
    "description": {
       "type": "string"
    },
    "email" : {
      "format": "email"
    },
  "dependencies": {
    "description": ["email"]
  }
};
```

And the data to be validated

``` javascript
var data = { description: "dfd"};
```

The result of the validation is 

``` javascript
{
  dataPath: ""
  keyword: "dependencies"
  message: "should have property email when property description is present"
}
```

I need dataPath to be able to associate error with some property and highlight component in UI.

Thanks.

**Comment by epoberezkin** [comment]
`dataPath` is empty because the validation error for dependencies keyword happens on the top level.

At the moment there is some inconsistency with how the errors are reported for `additionalProperties` and `required` keyword - instead of reporting the level at which the validation failed, they report the additional (or missing) property in `dataPath`. See #11, #18 and #55.

Currently you can use `{i18n: true}` option as a workaround - it will make both properties available in `params` property of the error object.

Your example in node REPL: https://tonicdev.com/esp/56317ddbd8fa530d00148e54

Please note that at the moment it will add all properties that should be present, not only those that are missing, so if you need to have more than one dependency it may not be what you need. I plan to address it soon in version 2.0 (#69)

**Comment by jsdmc** [comment]
Aha, I got it. Thanks. Maybe issue can be closed and referenced?

**Comment by epoberezkin** [comment]
No problem, let's keep it open until it's resolved in v2 or maybe earlier. And thanks for the feedback :)

**Comment by oleksandr-kupenko** [comment]
For 8 years, it was not possible to fix the problem? Why is dataPath needed if it's always empty?

**Comment by epoberezkin** [comment]
It’s not something to fix. Dependencies is a keyword that applies to the object, not to the property. So the errors are reported on the object level, and the properties are reported inside additional data in these errors.

I understand that you might need to translate it to the errors that will be shown to the end users - this has to be done in the application code or with ajv-errors package.

**Comment by epoberezkin** [comment]
dataPath is not always empty - it will be used for errors with property values and for any errors deeper in the data structure.

**Comment by oleksandr-kupenko** [comment]
> dataPath is not always empty - it will be used for errors with property values and for any errors deeper in the data structure.

Thank you. I messed up from "instancePath". Posted to the wrong branch (https://github.com/ajv-validator/ajv/issues/2237). But it's probably just an old name. And there I already found the answer that this is normal behavior.

I just also have an error in property. When the user makes a typo in a property that doesn't match the schema, I get an object with the error text but no instancePath. This is most likely what you meant when you said that you understand the need.
